apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -193,10 +195,23 @@ def load(cls, path, version=None, **kwargs):
             document = container.get_latest(max_version=version)
         except KeyError:
             document = cls(container, None, Dict(), version)
+            document.initialize_data()
+            document.validate()
         if version is not None and document.version < version:
             document = document.upgrade(version)
         return document
 
+    @classmethod
+    def create(cls, filename):
+        document = cls.load(filename, allow_missing=True)
+        document.save(overwrite=False)
+        return document
+
+    def initialize_data(self):
+        vkey = self.version_key
+        if vkey:
+            self._data[vkey] = str(self.version)
```

Comment:
Hmm.. This probably should be `self[vkey] = str(self.version)`, so that the value in `self._work` is also set.

Reply:
what is self._work? haven't used it for anything so far

Reply:
Changes dict contains three data storages: data, work and defaults. Work is merge between defaults and data. In addition, work is the dictionary which effects output from __iter__ and so on. The work dict is what is seen when using a ChangesDict and the data is what goes to disk. `ChangesDict.__setitem__()` write value to both data and work, so the data is stored on disk and is also the value used. In contrast, `ChangesDict.setdefault()` updates defaults and also work only if the key doesn't exists there.